### DISCUSSION TITLE
How to read inner Node/Attribute value

### DISCUSSION THREAD

**Discussion by stckwzrd** [zeux/pugixml#624] [General]
The main node is UANodeSet.
I can read the DataTypes, but not the "I" values. What should i do to be able to read them?
I use that code, to read subnodes: 
======================================
 pugi::xml_node tools = doc.child("UANodeSet").child("Aliases");
 for (pugi::xml_node_iterator it = tools.begin(); it != tools.end(); ++it)
 {
     std::cout << "==Aliases:";

     for (pugi::xml_attribute_iterator ait = it->attributes_begin(); ait != it->attributes_end(); ++ait)
     {
         std::cout << " " << ait->name() << "=" << ait->value();
     }

     std::cout << std::endl;
======================================
I have such xml:
![image](https://github.com/user-attachments/assets/e26e21e2-c565-4434-9492-5925c47075d1)

**Comment by zeux** [comment]
To read the "i=47" string you'd use `it->child_value()` or, equivalently, `it->text().get()`. Because it's a string that's not a number, you'd need to extract the number 47 if you are interested in it yourself.

**Reply by stckwzrd** [reply]
Thank you very much for your answer! It helped a lot!
